internal/docstore/query.go

```diff
@@ -0,0 +1,132 @@
+// Copyright 2019 The Go Cloud Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package docstore
+
+import (
+	"context"
+	"errors"
+	"io"
+	"strings"
+
+	"gocloud.dev/internal/docstore/driver"
+)
+
+// Query represents a query over a collection.
+type Query struct {
+	coll *Collection
+	dq   *driver.Query
+	err  QueryError
+}
+
+// Query creates a new Query over the collection.
+func (c *Collection) Query() *Query {
+	return &Query{coll: c, dq: &driver.Query{}}
+}
+
+// Where expresses a condition on the query.
+// Valid ops are: "=", ">", "<", ">=", "<=".
+func (q *Query) Where(fieldpath, op string, value interface{}) *Query {
+	fp, err := parseFieldPath(FieldPath(fieldpath))
+	if err != nil {
+		q.err = append(q.err, err)
+		return q
+	}
+	q.dq.Filters = append(q.dq.Filters, driver.Filter{
+		FieldPath: fp,
+		Op:        op,
+		Value:     value,
+	})
+	return q
+}
+
+// Limit will limit the results to at most n documents.
+func (q *Query) Limit(n int) *Query {
+	q.dq.Limit = n
+	return q
+}
+
+// Get returns an iterator for retrieving the documents specified by the query. If
+// field paths are provided, only those paths are set in the resulting documents.
+//
+// Call Stop on the iterator when finished.
+func (q *Query) Get(ctx context.Context, fieldpaths ...string) *DocumentIterator {
+	for _, fp := range fieldpaths {
+		fp, err := parseFieldPath(FieldPath(fp))
```

Comment:
Change the arg type to FieldPath, as in ActionList.Get.

Reply:
Done.